pkg/serverless/protocol.go

```diff
@@ -93,6 +101,10 @@ func (d *Daemon) UseAdaptiveFlush(enabled bool) {
 // In some circumstances, it may switch to another flush strategy after the flush.
 // shutdown indicates whether this is the last flush before the shutdown or not.
 func (d *Daemon) TriggerFlush(ctx context.Context, shutdown bool) {
+	// Increment the invocation wait group which tracks whether work is in progress for the daemon
+	d.InvcWg.Add(1)
+	defer d.InvcWg.Done()
```

Comment:
Would that make sense to use `StartInvocation` and `FinishInvocation` for clarity here?

Reply:
I can add a comment here. This case is a little different, because it we are trying to prevent the extension from moving on to the next event while flushing is happening. It doesn't necessarily match when the invocation is started or finished.

Reply:
OK got it, I thought it was something like that. Looks good to me then.